src/Tools/BuildValidator/CompilationDiff.cs

```diff
@@ -26,44 +37,153 @@ internal class CompilationDiff
             OriginalPath = originalPath;
         }
 
-        private CompilationDiff(
-            string originalPath,
-            Exception exception)
-        {
-            OriginalPath = originalPath;
-            Exception = exception;
-        }
-
         private CompilationDiff(ImmutableArray<Diagnostic> diagnostics, string originalPath)
         {
             Diagnostics = diagnostics;
             OriginalPath = originalPath;
         }
 
-        public static CompilationDiff Create(FileInfo assemblyFile, Compilation producedCompilation)
+        public static unsafe CompilationDiff Create(
+            FileInfo originalBinaryPath,
+            CompilationOptionsReader optionsReader,
+            Compilation producedCompilation,
+            IMethodSymbol? debugEntryPoint,
+            ILogger logger,
+            Options options)
         {
-            using var peStream = new MemoryStream();
+            using var rebuildPeStream = new MemoryStream();
 
-            var emitResult = producedCompilation.Emit(peStream);
-            if (emitResult.Success)
-            {
-                using var originalStream = assemblyFile.OpenRead();
-                var originalBytes = new byte[originalStream.Length];
-                originalStream.Read(originalBytes, 0, (int)originalStream.Length);
+            using var win32ResourceStream = producedCompilation.CreateDefaultWin32Resources(
```

Comment:
Can you add a comment as to why we are doing this. Essentially "by default command line adds a resource that we need to replicate manually here".

Reply:
sounds good. BTW, should we try extracting and copying over this resource from the original assembly?